File: glyph/src/main/java/com/hsj86715/ingress/glyph/pages/GlyphBaseAdapter.java
package com.hsj86715.ingress.glyph.pages;

import android.view.View;

import com.hsj86715.ingress.glyph.view.BaseRecyclerAdapter;
import com.hsj86715.ingress.glyphres.data.GlyphInfo;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by hushujun on 2017/5/16.
 * @author hushujun
 */

public class GlyphBaseAdapter extends BaseRecyclerAdapter {
    private List<GlyphInfo> mGlyphInfos;

    public GlyphBaseAdapter(List<GlyphInfo> glyphInfos) {
        if (glyphInfos == null) {
            mGlyphInfos = new ArrayList<>();
        } else {
            mGlyphInfos = glyphInfos;
        }
    }

    public GlyphBaseAdapter() {

    }

    public void setGlyphCategory(List<GlyphInfo> glyphInfos) {
        mGlyphInfos = glyphInfos;
        notifyDataSetChanged();
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, final int position) {
        final GlyphInfo info = mGlyphInfos.get(position);
        holder.glyphPreView.setSequences(info);
        if (mClickListener != null) {
            holder.itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mClickListener.onSequenceClicked(info);
                }
            });
        }
    }

    @Override
    public int getItemCount() {
        if (mGlyphInfos == null) {
            return 0;
        }
        return mGlyphInfos.size();
    }
}

File: app/src/main/java/com/hsj86715/ingress/glyph/pages/GlyphBaseAdapter.kt
package com.hsj86715.ingress.glyph.pages

import com.hsj86715.ingress.glyph.tools.Utils
import com.hsj86715.ingress.glyph.view.BaseRecyclerAdapter
import com.hsj86715.ingress.glyphres.data.BaseGlyphData
import com.hsj86715.ingress.glyphres.data.BaseGlyphData.Category

/**
 * Created by hushujun on 16/5/17.
 */
class GlyphBaseAdapter : BaseRecyclerAdapter() {
    private var mGlyphs: Array<Map.Entry<String, IntArray>>? = null

    fun setGlyphCategory(@Category category: String) {
        mGlyphs = arrayOf<Map.Entry<String, IntArray>>()
        mGlyphs = BaseGlyphData.getInstance().getGlyphByCategory(category).entries.toTypedArray<Map.Entry<String, IntArray>>()
        notifyDataSetChanged()
    }

    //position must be final
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val holderPos = position
        holder.glyphPreView.setSequences(mGlyphs!![holderPos].key)
        holder.itemView.setBackgroundColor(Utils.stringToColor(mGlyphs!![holderPos].key))
        holder.itemView.setOnClickListener { mClickedListener!!.onSequenceClicked(mGlyphs!![holderPos].key) }
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemCount(): Int {
        return mGlyphs!!.size
    }
}
